### ISSUE TITLE
Try to get minikube client again on certs error

### ISSUE THREAD

**Issue by priyawadhwa** [GoogleContainerTools/skaffold#4749]
We have an issue for skaffold + minikube, and the logs look like this:

```
Running "Run/Debug on Kubernetes" configuration (.vscode/launch.json) in Run mode
Running: skaffold dev -v info --port-forward --rpc-http-port 40051 --filename skaffold.yaml
starting gRPC server on port 50051
Using kubectl context: minikube
Listing files to watch...
 - go-hello-world
List generated in 6.066462ms
Generating tags...
 - go-hello-world -> go-hello-world:latest
Checking cache...
Tags generated in 105.186µs
 - go-hello-world: Found Locally
Cache check complete in 10.737784ms
Tags used in deployment:
 - go-hello-world -> go-hello-world:a85f9e4910ec6b4a1dfe32cb0b72c6824ca37b8a2544517506b1424cd45c38d7
exiting dev mode because first deploy failed: unable to connect to Kubernetes: Get "https://[REDACTED_IP_ADDRESS]/version?timeout=32s": x509: certificate is valid for 172.18.0.2, 10.96.0.1, 127.0.0.1, 10.0.0.1, not 172.18.0.3
Exited with code 1.
To disable cleaning up resources, set cleanUp to false in your launch configuration '/home/grappeggia/cloudcode-projects/go-hello-world-2/.vscode/launch.json'
Cleaning up...
reading manifests: kubectl create: running [kubectl --context minikube create --dry-run=client -oyaml -f /home/grappeggia/cloudcode-projects/go-hello-world-2/kubernetes-manifests/hello.deployment.yaml -f /home/grappeggia/cloudcode-projects/go-hello-world-2/kubernetes-manifests/hello.service.yaml]
 - stdout: ""
 - stderr: "Unable to connect to the server: x509: certificate is valid for 172.18.0.2, 10.96.0.1, 127.0.0.1, 10.0.0.1, not 172.18.0.3\n"
 - cause: exit status 1
```

As discussed w/ @nkubala and @medyagh, this bug is happening because skaffold runs $(minikube docker-env) at the beginning of the dev loop and stores the env variables in memory. However, if minikube's IP changes, then those env variables change as well. 

If skaffold can try to get the minikube client again, and rerun $(minikube docker-env), when it sees this error, it should prevent it from happening again in the future.

**Comment by briandealwis** [comment]
Out of curiosity, what can lead to the IP address change?

**Comment by priyawadhwa** [comment]
Usually it happens on restart, if a user runs `minikube stop` and then `minikube start` again, but something like closing your laptop can also do it.

**Comment by tstromberg** [comment]
I was a bit confused by this error report, and now I'm even further confused. The error is raised by kubectl, not Skaffold. 

This particular error is out of Skaffold's control: the context written to `$HOME/.kube/config` no longer matches the expectations of the cluster. For instance, the cluster is in the middle of a restart

I believe this may have been fixed in minikube v1.13.1 - https://github.com/kubernetes/minikube/pull/9143/files

There is still an underlying issue where Skaffold is not aware of IP changes in minikube, but that would result in a different error, talking to a Docker daemon with an invalid certificate.
